contracts-cli/index.js

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env node
+
+const Web3 = require('web3')
+const fs = require('fs')
+const path = require('path')
+const Accounts = require('web3-eth-accounts')
+const HDWalletProvider = require('truffle-hdwallet-provider')
+const HDWalletProviderPK = require('truffle-hdwallet-provider-privkey')
+const ProviderEngine = require('web3-provider-engine')
+const RpcSubprovider = require('web3-provider-engine/subproviders/rpc')
+const LedgerWalletSubproviderFactory = require('ledger-wallet-provider').default
+const { prompt } = require('inquirer')
+
+const config = require('./config')
+const contract = require('./contract')
+const { log, capitalize } = require('./cli-utils')
+
+const LEDGER_ROOT_PATH = `44'/60'/0'/0`
+
+var web3
+var ethNetwork = 'testnet' // testnet or mainnet
+var account
+
+const determineNetwork = async () => {
+  const answers = await prompt([{
+    type: 'list',
+    name: 'network',
+    message: 'Select Ethereum network: ',
+    choices: ['testnet', 'mainnet']
+  }])
+  ethNetwork = answers.network
+}
+
+const requestPrivateKey = async () => {
+  const answers = await prompt([{
+    type: 'password',
+    name: 'privateKey',
+    message: 'Enter private key: ',
+  }])
+  return answers.privateKey
+}
+
+const requestMnemonic = async () => {
+  const answers = await prompt([{
+    type: 'password',
+    name: 'mnemonic',
+    message: 'Enter mnemonic passphrase: ',
+  }])
+  return answers.mnemonic
+}
+
+const requestAccountSelection = async (accounts) => {
+  const answers = await prompt([{
+    type: 'list',
+    name: 'account',
+    message: 'Select ETH account: ',
+    choices: accounts,
+  }])
+  return accounts.indexOf(answers.account)
+}
+
+const decodeKeystore = async() => {
+  const answers = await prompt([
+    {
+      type: 'input',
+      name: 'keystorePath',
+      message: 'Enter path of keystore file: ',
+    },
+    {
+      type: 'input',
+      name: 'password',
+      message: 'Enter password to decrypt file: ',
+    }
+  ])
+  const file = JSON.parse(fs.readFileSync(answers.keystorePath))
```

Comment:
Could use https://github.com/jprichardson/node-fs-extra/blob/master/docs/readJson.md to be async considering the function itself is async.